datapy/Table.py

```diff
@@ -0,0 +1,178 @@
+    
+    @staticmethod
+    def db_session(table_root):
+        """ Establish a connection with a sqlite database
+
+        If a file path is given, returns a connection to that database file,
+        otherwise returns a connection to an in-memory database.
+
+        Args:
+            self.table_root (Optional[str], optional): If provided, a file path to a
+            persistent database file. Defaults to None.
+
+        Returns:
+            sql.Connection: connection to the database
+        """
+        if os.path.isfile(table_root):
+            conn = sql.connect(":memory:")
+            sql_file = open(table_root)
+            sql_string = sql_file.read()
+            conn.executescript(sql_string)
+            return conn
+        return sql.connect(":memory:")
+    
+    def create_table(self, conn, headers):
+        """ Create a table given the headers
+        Args:
+            self.table_name (str): name of the table to create
+            conn (sql.Connection): connection to the database
+            table_header (str): table headers in string format -
+                                "field::type,field::type"
+
+        Returns:
+            bool: Was the table created?
+        """
+        headers = [h.replace("::", " ") for h in headers.split(",")]
+        create_stmt = f"create table if not exists {self.table_name} ( {', '.join(headers)} )"
+        c = conn.cursor()
+        try:
+            c.execute(create_stmt)
+            return True
+        except:
+            return False
+
+    def load_table(self, conn):
+        """ loads tables from a previous db dump
+        Args:
+            self.table_root (str): path to table
+            conn (sql.Connection): connection to the database
+        """
+        c = conn.cursor()
+        
+    def ingest_csv(self, conn):
+        """ Read a properly formatted .csv file and stash it in a sqlite table
+        Args:
+            self.input_data (str): path to the .csv file
+            conn (sql.Connection): connection to the database
+        Returns:
+            int: the number of table rows inserted
+        """
+        with open(self.input_data) as f, conn:
+            headers = f.readline().strip()
+            created = self.create_table(conn, headers)
+            if created:
+                rows_inserted = self.insert_rows(conn, headers, f)
+            try:
+                conn.commit()
+                return rows_inserted
+            except:
+                sys.exit("No rows were inserted, error")
+
+    def insert_rows(self, conn, headers, file_conn):
+        """ Insert rows from a csv file connection into a database table
+        Args:
+            self.table_name (str): name of the table to insert records into
+            conn (sql.Connection): connection to the database
+            headers (str): table headers in string format -
+                           "field::type,field::type"
+            file_conn (TextIOWrapper): file connection
```

Comment:
I've never seen Python use file connection objects. This seems a bit out of place (mighty Java-like)

Reply:
This is the text stream opened in with statement on line 112